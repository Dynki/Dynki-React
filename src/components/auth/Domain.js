import React, { Component } from 'react';
import { connect } from 'react-redux';
import { debounce } from 'lodash';

import DomainForm from './DomainForm';
import { checkDomain, createDomain } from '../../store/actions/domainActions';

class Domain extends Component {
    constructor(props) {
        super(props);
        this.onCheckDomain = debounce(this.onCheckDomain, 1000)
    }

    // Call logic to determine this this domain name is valid and unique.
    onCheckDomain = (name) => {
        this.props.CheckDomain(name);
    }

    // Fire logic to persist the new domain name.
    onCreateDomain = (name) => {
        this.props.CreateDomain(name);
    }
      
    render() {
        const {...domain } = this.props.domain;
 
        return (
            <div className="login">
                <h1 className="registration__heading">Name your team</h1>
                <h4>Give your team a name they can be proud of</h4>
                <DomainForm {...domain} onCheckDomain={this.onCheckDomain} onCreateDomain={this.onCreateDomain}></DomainForm>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return{
      authError: state.auth.authError,
      pending: state.auth.pending,
      auth: state.firebase.auth,
      domain: state.domain
    }
  }
  
const mapDispatchToProps = (dispatch) => {
    return {
        CheckDomain: (name) => dispatch(checkDomain(name)),
        CreateDomain: (name) => dispatch(createDomain(name))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Domain);