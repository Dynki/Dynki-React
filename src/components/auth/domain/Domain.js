import React from 'react';
import styles from 'styled-components'

import DomainForm from './DomainForm';

const StyledForm = styles.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    align-items: center;

    min-height: 378px;
    background-color: #ffffff;
    padding: 36px!important;
    width: 100%;
    height: 600px;

    .heading {
        color:  #3095de;;
        // font-family: Raleway;
        font-size: 37px;
        margin-bottom: 20px;
    }

    nz-form-item {
        width: 100%;
    }

    form {
        min-width: 100%;
    }

    .button {
        height: 73px;
        font-size: 35px;
        width: 100%;
    
        span {
            padding-right: 30px;
        }
    }
`;

const Domain = ({ checkDomain, updateDomain, domain, pending }) => {

    // Fire logic to persist the new domain name.
    const onUpdateDomain = name => {
        updateDomain(name);
    }
      
    return (
        <StyledForm>
            <h1 className="registration__heading">Name your team</h1>
            <h4>Give your team a name they can be proud of</h4>
            <DomainForm onUpdateDomain={onUpdateDomain} pending={pending}></DomainForm>
        </StyledForm>
    );
}

export default Domain;