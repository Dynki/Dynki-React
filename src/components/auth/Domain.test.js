import React from 'react';
import { shallow } from 'enzyme';
import { Domain, mapDispatchToProps, mapStateToProps } from './Domain';

it('should render full component when domain has been checked', () => {
    const wrapper = shallow(<Domain />);
    expect(wrapper).toMatchSnapshot();
});

it('should call check domain with correct name', done => {

    const mockCheckDomain = jest.fn();
    const wrapper = shallow(<Domain CheckDomain={mockCheckDomain}/>);
    wrapper.instance().onCheckDomain('TestName');

    setTimeout(() => {
        expect(wrapper.instance().props.CheckDomain).toHaveBeenCalledTimes(1);
        done();
    }, 1000);
});

it('should call crate domain with correct name', () => {
    const mockCreateDomain = jest.fn();
    const wrapper = shallow(<Domain CreateDomain={mockCreateDomain}/>);
    wrapper.instance().onCreateDomain('TestName');
    expect(wrapper.instance().props.CreateDomain).toHaveBeenCalledTimes(1);
});

it('should set up state', () => {
    const initialState = {
        auth: { authError: null, pending: false },
        firebase: { auth: undefined },
        domain: 'DOM123'
    }
  
    expect(mapStateToProps(initialState).authError).toEqual(initialState.auth.authError);
    expect(mapStateToProps(initialState).pending).toEqual(initialState.auth.pending);
    expect(mapStateToProps(initialState).auth).toEqual(initialState.firebase.auth);
    expect(mapStateToProps(initialState).domain).toEqual(initialState.domain);
});

it('should set up dispatch', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).CheckDomain();
    mapDispatchToProps(dispatch).CreateDomain();
    expect(dispatch).toHaveBeenCalledTimes(2);
});