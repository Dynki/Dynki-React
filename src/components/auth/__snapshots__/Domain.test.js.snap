// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render full component when domain has been checked 1`] = `
<div
  className="login"
>
  <h1
    className="registration__heading"
  >
    Name your team
  </h1>
  <h4>
    Give your team a name they can be proud of
  </h4>
  <DomainForm
    onCheckDomain={[Function]}
    onCreateDomain={[Function]}
  />
</div>
`;
